diff -crNB blocks-new/packages/file-storage/tcl/file-storage-callback-procs.tcl blocks-clean/packages/file-storage/tcl/file-storage-callback-procs.tcl
*** blocks-new/packages/file-storage/tcl/file-storage-callback-procs.tcl	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/tcl/file-storage-callback-procs.tcl	2009-07-05 21:25:19.000000000 -0400
***************
*** 243,245 ****
--- 243,367 ----
  	
  	
      }      
+ 
+ ad_proc -public -callback search::url -impl content_extlink {} {
+     returns a url from an object in the file storage to the search package
+ } {
+     set item_id $object_id
+     set ext_url [db_string get_url { *SQL* } -default ""]
+ 
+     return "$ext_url"
+ }
+ 
+ ad_proc -public -callback search::url -impl content_folder {} {
+     returns a url from a folder in the file storage to the search package
+ } {
+     set folder_id $object_id
+     set folder_data [fs::get_folder_package_and_root $folder_id]
+     set package_id [lindex $folder_data 0]
+     set package_url [apm_package_url_from_id $package_id]
+ 
+     return "${package_url}?folder_id=$folder_id"
+ }
+ 
+ ad_proc -public -callback planner::edit_url -impl content_folder {} {
+     returns a url to the edit action of a folder in the file storage to the search package
+ } {
+     set folder_id $object_id
+     set folder_data [fs::get_folder_package_and_root $folder_id]
+     set package_id [lindex $folder_data 0]
+     set package_url [apm_package_url_from_id $package_id]
+ 
+     return "${package_url}folder-edit?folder_id=$folder_id"
+ }
+ 
+ ad_proc -public -callback planner::edit_url -impl file_storage_object {} {
+     returns a url to the edit action of an object in the file storage to the search package
+ } {
+     set item_id [content::revision::item_id -revision_id $object_id]
+     set package_id [acs_object::package_id -object_id $item_id]
+     set package_url [apm_package_url_from_id $package_id]
+ 
+     return "${package_url}file?file_id=$item_id"
+ }
+ 
+ ad_proc -public -callback planner::edit_url -impl content_extlink {} {
+     returns a url to the edit action of an object in the file storage to the search package
+ } {
+     set item_id $object_id
+     set package_id [acs_object::package_id -object_id $item_id]
+     set package_url [apm_package_url_from_id $package_id]
+ 
+     return "${package_url}simple-edit?object_id=$item_id"
+ }
+ 
+ ad_proc -public -callback planner::delete_url -impl content_folder {} {
+         returns a url to the delete action of a folder in the file storage to the search package
+ } {
+     set folder_id $object_id
+     set folder_data [fs::get_folder_package_and_root $folder_id]
+     set package_id [lindex $folder_data 0]
+     set package_url [apm_package_url_from_id $package_id]
+ 
+     return "${package_url}folder-delete?folder_id=${folder_id}&form:id=folder-delete"
+ }
+ 
+ ad_proc -public -callback planner::delete_url -impl file_storage_object {} {
+         returns a url to the delete action of an object in the file storage to the search package
+ } {
+     set item_id [content::revision::item_id -revision_id $object_id]
+     set package_id [acs_object::package_id -object_id $item_id]
+     set package_url [apm_package_url_from_id $package_id]
+ 
+     return "${package_url}delete?object_id=${item_id}&form:id=delete_confirm"
+ }
+ 
+ ad_proc -public -callback planner::delete_url -impl content_extlink {} {
+         returns a url to the delete action of an object in the file storage to the search package
+ } {
+     set item_id $object_id
+     set folder_id [db_string get_folder_id { *SQL* } -default 0]
+     set package_id [acs_object::package_id -object_id $item_id]
+     set package_url [apm_package_url_from_id $package_id]
+ 
+     return "${package_url}simple-delete?object_id=${item_id}&folder_id=$folder_id"
+ }
+ 
+ ad_proc -public -callback planner::insert_object -impl file-storage {} {
+ 
+     @author alvaro@viaro.net
+     @creation_date Jul-2008
+ 
+     Insert the recently created object into the blocks view objects table
+ 
+ } {
+     set block_id [ad_get_cookie fs_block_id "0"]
+     if { $block_id != 0 && [ad_get_cookie fs_community_id ""] == $community_id } {
+         set object_index [planner::get_next_object_index -block_id $block_id]
+         planner::insert_object_to_block -block_id $block_id -object_id $object_id -object_index $object_index
+         ad_set_cookie fs_block_id 0
+         set community_id [planner::get_community_id_from_block_id -block_id $block_id]
+         set community_url [dotlrn_community::get_community_url $community_id]
+         planner::show_alert_message -object_id $object_id -community_url $community_url
+     } else {
+         set community_id [dotlrn_community::get_community_id]
+     }
+     planner::flush_blocks_cache -community_id $community_id
+ }
+ 
+ ad_proc -public -callback planner::extend_form -impl file-storage {} {
+ 
+     @author alvaro@viaro.net
+     @creation_date Jul-2008
+ 
+     Extend the form of the file-storage to add items to blocks view table
+ 
+ } {
+     upvar elements elements
+     set block_id [ad_get_cookie fs_block_id "0"]
+     set community_url [dotlrn_community::get_community_url $community_id]
+     set display_t "<a class=\"button\" href=\"[set community_url]planner/object-add?object_id=@contents.object_id@\">\#planner.blocks_choose\#</a>"
+     if { $block_id != 0 && [ad_get_cookie fs_community_id 0] == $community_id} {
+         lappend elements choose_link [list label "" display_template $display_t]
+     }
+ }
diff -crNB blocks-new/packages/file-storage/www/delete.tcl blocks-clean/packages/file-storage/www/delete.tcl
*** blocks-new/packages/file-storage/www/delete.tcl	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/delete.tcl	2009-07-05 21:29:52.000000000 -0400
***************
*** 82,87 ****
--- 82,90 ----
  	    }
  	}
      }
+ ## Planner View -> Flush Cache
+     callback -catch -impl "planner" planner::flush_blocks_cache -community_id [dotlrn_community::get_community_id]
+ ###
      ad_returnredirect $return_url
      ad_script_abort
  }
diff -crNB blocks-new/packages/file-storage/www/file-add.tcl blocks-clean/packages/file-storage/www/file-add.tcl
*** blocks-new/packages/file-storage/www/file-add.tcl	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/file-add.tcl	2009-07-05 21:29:03.000000000 -0400
***************
*** 249,254 ****
--- 249,257 ----
              set file_id [db_nextval "acs_object_id_seq"]
          }
      }
+ ## Planner View
+     callback -catch -impl "file-storage" planner::insert_object -block_id 0 -object_id $this_file_id -community_id [dotlrn_community::get_community_id]
+ ###
      file delete $upload_file.tmpfile
  } -edit_data {
      set this_title $title
diff -crNB blocks-new/packages/file-storage/www/file.adp blocks-clean/packages/file-storage/www/file.adp
*** blocks-new/packages/file-storage/www/file.adp	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/file.adp	2009-07-05 21:28:37.000000000 -0400
***************
*** 3,8 ****
--- 3,9 ----
  <property name="header">@title;noquote@</property>
  <property name="context">@context;noquote@</property>
  <property name="displayed_object_id">@file_id;noquote@</property>
+ <property name="planner_object_id">@file_id;noquote@</property>
  
  <h1>@title;noquote@ - #file-storage.properties#</h1>
  
diff -crNB blocks-new/packages/file-storage/www/file-edit.adp blocks-clean/packages/file-storage/www/file-edit.adp
*** blocks-new/packages/file-storage/www/file-edit.adp	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/file-edit.adp	2009-07-05 21:30:41.000000000 -0400
***************
*** 2,7 ****
--- 2,8 ----
  <property name="title">@page_title;noquote@</property>
  <property name="header">@page_title;noquote@</property>
  <property name="context">@context;noquote@</property>
+ <property name="planner_object_id">@file_id;noquote@</property>
  
  <formtemplate id="file-edit"></formtemplate>
  
diff -crNB blocks-new/packages/file-storage/www/file-edit.tcl blocks-clean/packages/file-storage/www/file-edit.tcl
*** blocks-new/packages/file-storage/www/file-edit.tcl	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/file-edit.tcl	2009-07-05 21:30:14.000000000 -0400
***************
*** 68,73 ****
--- 68,76 ----
  								       -container_object_id $package_id \
  								       -element_name category_id]
      }
+ ## Planner View -> Flush Cache
+     callback -catch -impl "planner" planner::flush_blocks_cache -community_id [dotlrn_community::get_community_id]
+ ###
  } -after_submit {
      ad_returnredirect "file?file_id=$file_id"
  }
diff -crNB blocks-new/packages/file-storage/www/folder-chunk.tcl blocks-clean/packages/file-storage/www/folder-chunk.tcl
*** blocks-new/packages/file-storage/www/folder-chunk.tcl	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/folder-chunk.tcl	2009-07-05 21:31:28.000000000 -0400
***************
*** 148,153 ****
--- 148,157 ----
  		       link_html { title "[_ file-storage.Download]" }] \
  		  ]
  
+ ## Planner View -> Extend Form
+ if {[exists_and_not_null return_url]} {
+     callback -catch -impl "file-storage" planner::extend_form -block_id 0 -referrer $return_url -community_id [dotlrn_community::get_community_id]
+ }
  
  if { $categories_p } {
      lappend elements categories [list label [_ file-storage.Categories] display_col "categories;noquote"]
***************
*** 167,172 ****
--- 171,181 ----
  
  if {$allow_bulk_actions} {
      set bulk_actions [list "[_ file-storage.Move]" "${fs_url}move" "[_ file-storage.lt_Move_Checked_Items_to]" "[_ file-storage.Copy]" "${fs_url}copy" "[_ file-storage.lt_Copy_Checked_Items_to]" "[_ file-storage.Delete]" "${fs_url}delete" "[_ file-storage.Delete_Checked_Items]" "[_ file-storage.Download_ZIP]" "${fs_url}download-zip" "[_ file-storage.Download_ZIP_Checked_Items]"]
+ ## Planner View -> Add the SELECT bulk action when coming from dotlrn blocks
+     if { [ad_get_cookie fs_block_id 0] != 0 && [ad_get_cookie fs_community_id 0] != 0 } {
+         lappend bulk_actions "[_ planner.select_multiple]" "[dotlrn_community::get_community_url [dotlrn_community::get_community_id]]planner/object-add" "[_ file-storage.select_multiple]"
+     }
+ ###
      callback fs::folder_chunk::add_bulk_actions \
  	-bulk_variable "bulk_actions" \
  	-folder_id $folder_id \
diff -crNB blocks-new/packages/file-storage/www/folder-create.tcl blocks-clean/packages/file-storage/www/folder-create.tcl
*** blocks-new/packages/file-storage/www/folder-create.tcl	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/folder-create.tcl	2009-07-05 21:32:29.000000000 -0400
***************
*** 113,118 ****
--- 113,121 ----
  								       -element_name category_id]
      }
  
+ ## Planner View
+     callback -catch -impl "file-storage" planner::insert_object -block_id 0 -object_id $folder_id -community_id [dotlrn_community::get_community_id]
+ ###
      ad_returnredirect "?folder_id=$folder_id"
      ad_script_abort
  } -edit_data {
diff -crNB blocks-new/packages/file-storage/www/folder-edit.tcl blocks-clean/packages/file-storage/www/folder-edit.tcl
*** blocks-new/packages/file-storage/www/folder-edit.tcl	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/folder-edit.tcl	2009-07-05 21:33:28.000000000 -0400
***************
*** 54,59 ****
--- 54,62 ----
      }
  
      callback fs::folder_edit -package_id [ad_conn package_id] -folder_id $folder_id
+ ## Planner View -> Flush Cache
+     callback -catch -impl "planner" planner::flush_blocks_cache -community_id [dotlrn_community::get_community_id]
+ ###
  
  } -after_submit {
      ad_returnredirect "?folder_id=$folder_id"
diff -crNB blocks-new/packages/file-storage/www/index.tcl blocks-clean/packages/file-storage/www/index.tcl
*** blocks-new/packages/file-storage/www/index.tcl	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/index.tcl	2009-07-08 14:43:30.000000000 -0400
***************
*** 28,33 ****
--- 28,41 ----
      context:onevalue
  }
  
+ ## Planner View 
+ set urls [split $return_url "/"]
+ if { ![regexp "planner\??.*" [lindex $urls [expr [llength $urls] - 2]]] } {
+     ad_set_cookie fs_block_id 0
+     ad_set_cookie fs_community_id 0
+ }
+ ###
+ 
  # Don't allow delete if root folder
  set root_folder_p 0
  set root_folder_id [fs_get_root_folder]
diff -crNB blocks-new/packages/file-storage/www/simple-add.tcl blocks-clean/packages/file-storage/www/simple-add.tcl
*** blocks-new/packages/file-storage/www/simple-add.tcl	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/simple-add.tcl	2009-07-05 21:38:00.000000000 -0400
***************
*** 99,106 ****
  
      fs::do_notifications -folder_id $folder_id -filename $url -item_id $item_id -action "new_url"
  
  } -after_submit {
  
      ad_returnredirect "?folder_id=$folder_id"
! 
  }
--- 99,109 ----
  
      fs::do_notifications -folder_id $folder_id -filename $url -item_id $item_id -action "new_url"
  
+ ## Planner View
+     callback -catch -impl "file-storage" planner::insert_object -block_id 0 -object_id $item_id -community_id [dotlrn_community::get_community_id]
+ ###
  } -after_submit {
  
      ad_returnredirect "?folder_id=$folder_id"
!     ad_script_abort
  }
diff -crNB blocks-new/packages/file-storage/www/simple.adp blocks-clean/packages/file-storage/www/simple.adp
*** blocks-new/packages/file-storage/www/simple.adp	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/simple.adp	2009-07-05 21:36:47.000000000 -0400
***************
*** 3,8 ****
--- 3,9 ----
  <property name="header">@title;noquote@</property>
  <property name="context">@context;noquote@</property>
  <property name="displayed_object_id">@object_id;noquote@</property>
+ <property name="planner_object_id">@object_id;noquote@</property>
  
  <p><a href="@url@">@url@</a></p>
  <p>@description@</p>
diff -crNB blocks-new/packages/file-storage/www/simple-delete.tcl blocks-clean/packages/file-storage/www/simple-delete.tcl
*** blocks-new/packages/file-storage/www/simple-delete.tcl	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/simple-delete.tcl	2009-07-05 21:42:11.000000000 -0400
***************
*** 22,27 ****
--- 22,30 ----
  
  }
  
+ ## Planner View -> Flush Cache
+     callback -catch -impl "planner" planner::flush_blocks_cache -community_id [dotlrn_community::get_community_id]
+ ###
  
  ad_returnredirect "./?folder_id=$folder_id"
  
diff -crNB blocks-new/packages/file-storage/www/simple-edit.tcl blocks-clean/packages/file-storage/www/simple-edit.tcl
*** blocks-new/packages/file-storage/www/simple-edit.tcl	2009-07-05 21:02:54.000000000 -0400
--- blocks-clean/packages/file-storage/www/simple-edit.tcl	2009-07-05 21:34:13.000000000 -0400
***************
*** 38,43 ****
--- 38,46 ----
  								       -container_object_id $package_id \
  								       -element_name category_id]
      }
+ ## Planner View -> Flush Cache
+     callback -catch -impl "planner" planner::flush_blocks_cache -community_id [dotlrn_community::get_community_id]
+ ###
      ad_returnredirect "?[export_vars folder_id]"
  }
  
