diff -crNB dotlrn-2.4.1/packages/theme-zen/lib/lrn-master.adp theme-zen/lib/lrn-master.adp
*** dotlrn-2.4.1/packages/theme-zen/lib/lrn-master.adp	2009-07-05 21:03:00.000000000 -0400
--- theme-zen/lib/lrn-master.adp	2009-07-07 14:00:38.000000000 -0400
***************
*** 84,89 ****
--- 84,92 ----
      <div id="sub-navigation">
        <div class="block-marker">#theme-zen.begin_sub_navigation#</div>
        @subnavbar;noquote@
+       <if @community_id@ ne "" and @planner_p@>
+         <include src="/packages/planner/lib/navigation" planner_object_id="@planner_object_id@">
+       </if>
      </div>
    </if>
  
diff -crNB dotlrn-2.4.1/packages/theme-zen/lib/lrn-master.tcl theme-zen/lib/lrn-master.tcl
*** dotlrn-2.4.1/packages/theme-zen/lib/lrn-master.tcl	2009-07-05 21:03:00.000000000 -0400
--- theme-zen/lib/lrn-master.tcl	2009-07-07 14:00:56.000000000 -0400
***************
*** 277,279 ****
--- 277,285 ----
      template::head::add_css -href "/resources/dotlrn/dotlrn-toolbar.css"
      template::add_header -src "/packages/dotlrn/lib/toolbar"
  }
+ ## Planner: if there is no planner_object_id set the variable
+ set planner_p [apm_package_installed_p "planner"]
+ if {![exists_and_not_null planner_object_id]} {
+     set planner_object_id 0
+ }
+ ###
